src/abstract_register.jl

```diff
@@ -163,44 +176,68 @@ end
 
 ## Measurement
 
+export ComputationalBasis
+struct ComputationalBasis end
```

Comment:
You can add a fall back to basis for measure functions here, so you don't have to define the fallback in ArrayReg.

Reply:
didn't get it

Reply:
Can I still use the following (if the original interface is kept)?

```julia
measure(register, locs...)
```

Then, the optional arg `op` won't work here, since you didn't define the optional, unless you define it here

```julia
@interface measure(op, register, locs...)
@interface measure(register, locs) # equilavent to = error("not implemented blabla")
```

missing `@interface measure(register, locs)` will cause inconsistent behaviour of errors and will make people to define it every time they define a custom register. (and this change will make me to change what's in YaoGraph as well), so I think it is better to keep `measure(register, locs)` and others as it is, and add an extra `measure(op, register, locs)`

like `Base.show`, you can just define `show(io, x)` if you don't actually want to consider others, but you can still  define `show(io, mimetype, x)` if you want to define several. so just define the fallbacks here so we don't need to define them in concrete packages (and we don't need to change a lot code while still being compatible)

```julia
measure(::ComputationalBasis, register, locs) = measure(register, locs)
```

Reply:
Let's keep it is for this moment, we can fine tune it later, since it will not touch the interface.